corehq/apps/registration/models.py

```diff
@@ -155,3 +156,36 @@ def clear_data_for_username(cls, username):
         :param username: string
         """
         cls.objects.filter(username=username).delete()
+
+
+class SelfSignupWorkflow(models.Model):
+    domain = models.CharField(max_length=255)
+    initiating_user = models.CharField(max_length=80)
+    created_on = models.DateTimeField(auto_now_add=True)
+    completed_on = models.DateTimeField(null=True)
+    subscribed_edition = models.CharField(
+        choices=SoftwarePlanEdition.CHOICES,
+        max_length=25,
+        null=True
+    )
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["domain", "completed_on"]),
```

Comment:
Under normal circumstances, a domain should only ever have one self-signup object, since the workflow will be initiated a the time of domain creation. So I'm not sure if the 'What are all the self signups for X domain' is of particular relevance -- unless indexing here would also help with retrieving the 0 or 1 self signups a domain should have (and it might, I'm not too familiar with indexing generally).

Reply:
I see. This does feel like a nitpick now, but if there should only ever be one self-signup per domain, then indexing the `completed_on` field seems unnecessary, since `domain` would give us the quick lookup for that one object. In addition, I could see wanting to make ad-hoc queries to find the status of a domain's self-signup, without regards to the `completed_on` field. So it's probably fine as is, but might be better to just index on `domain`

Reply:
Good point - domain is really the important thing here, so that's probably what makes sense to index on https://github.com/dimagi/commcare-hq/pull/35177/commits/41e0db5b51c5c3582e2d6b522de115b569e62ac3